hcp/handler.go

```diff
@@ -0,0 +1,33 @@
+package hcp
+
+import (
+	"context"
+
+	sdkpacker "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer/hcl2template"
+	"github.com/hashicorp/packer/internal/registry"
+	"github.com/hashicorp/packer/packer"
+)
+
+type HCPHandler interface {
+	Bucket() *registry.Bucket
+	PopulateIteration(context.Context) error
+	BuildStart(context.Context, string) error
+	BuildDone(ctx context.Context, buildName string, artifacts []sdkpacker.Artifact, buildErr error) ([]sdkpacker.Artifact, error)
+}
+
+// TrySetupHCP attempts to setup the HCP-related structures if HCP is enabled
```

Comment:
Comment needs to be updated. 
```suggestion
// GetHCPHandler .....
```